clash_lib/src/proxy/hysteria2/udp_hop.rs

```diff
@@ -0,0 +1,200 @@
+use std::{
+    fmt::Debug,
+    io,
+    net::SocketAddr,
+    ops::{Deref, DerefMut, Sub},
+    sync::{Arc, Mutex},
+    task::{Context, Poll},
+    time::{Duration, Instant},
+};
+
+use futures::ready;
+use quinn::{
+    udp::{Transmit, UdpSocketState, UdpState},
+    AsyncUdpSocket,
+};
+
+use tokio::{io::Interest, net::UdpSocket};
+
+use crate::proxy::converters::hysteria2::PortGenrateor;
+
+pub struct UdpHop {
```

Comment:
could you please add a short description for this struct, such as what part it plays in the protocol

Reply:
https://v2.hysteria.network/docs/advanced/Port-Hopping/